tools/check_code_style

```diff
@@ -31,5 +31,26 @@ fail() {
     exit 1
 }
 
-git grep -P 'check_screen.*00(?!.*nocheck:)' || exit 0
-fail "See https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/CONTRIBUTING.md#coding-style for more details"
+
+forbidden_check_var=0
+rules_violated=0
+files=$(git --no-pager diff --name-only HEAD $(git merge-base HEAD master))
+
+# rule 001
+for file in $files; do
+    if [[ -f $file   &&  $file == *.pm  && $file != *"Architectures.pm"*  &&  $file != *"Backends.pm"* ]]; then
+         grep -oHEn "check_var\('ARCH',.*\)|check_var\('BACKEND',.*\)" $file && ((forbidden_check_var+=1))
+    fi
+done
+if [[ $forbidden_check_var > 0 ]]; then
+	rules_violated+=1
+fi;
+
+# rule 002
+git grep -q 'check_screen.*00(?!.*nocheck:)' || (echo "Use of check_screen with zero timeout is forbidden" && rules_violated+=1)
+if  [[  $rules_violated > 0 ]]; then
+   if [[ $forbidden_check_var ]];  then
+     echo "Use defined ARCH or BACKEND functions instead of caling check_var() functions in above mentioned file with line number" 
+   fi
+   fail "See https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/CONTRIBUTING.md#coding-style for more details"
+fi
```

Comment:
Seems you are having a hard time counting if there were errors encountered or not. You might have an easier time adding your checks into t/01_style.t instead where you can use Test::More instructions

Reply:
Thanks Oliver, I added the checks into t/01_style. Please let me know is this ok ?

Reply:
looks cool, just a minor nit-pick regarding a double white-space :)